packages-exp/auth-exp/src/api/authentication/email_and_password.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Auth } from '../../model/auth';
+import { IdTokenResponse, IdToken } from '../../model/id_token';
+import {
+  performSignInRequest,
+  HttpMethod,
+  Endpoint,
+  performApiRequest
+} from '..';
+
+export interface SignInWithPasswordRequest {
+  returnSecureToken?: boolean;
+  email: string;
+  password: string;
+}
+
+export interface SignInWithPasswordResponse extends IdTokenResponse {
+  email: string;
+  displayName: string;
+}
+
+export async function signInWithPassword(
+  auth: Auth,
+  request: SignInWithPasswordRequest
+): Promise<SignInWithPasswordResponse> {
+  return performSignInRequest<
+    SignInWithPasswordRequest,
+    SignInWithPasswordResponse
+  >(auth, HttpMethod.POST, Endpoint.SIGN_IN_WITH_PASSWORD, request);
+}
+
+export enum GetOobCodeRequestType {
+  PASSWORD_RESET = 'PASSWORD_RESET',
+  EMAIL_SIGNIN = 'EMAIL_SIGNIN',
+  VERIFY_EMAIL = 'VERIFY_EMAIL',
+  VERIFY_AND_CHANGE_EMAIL = 'VERIFY_AND_CHANGE_EMAIL'
+}
+
+interface GetOobCodeRequest {
+  email?: string; // Everything except VERIFY_AND_CHANGE_EMAIL
+  // captchaResp?: string, // RESET_PASSWORD
```

Comment:
Why are these commented out?

Reply:
they were commented out in the auth-ts branch, let me clean it up now that we know the signature better

Reply:
should be fixed now